File: services/orders-service/src/main/java/pl/xsware/orders/domain/order/Currency.java
package pl.xsware.orders.domain.order;

public enum Currency {
    PLN,
    EUR,
    USD;

    public static Currency from(String value) {
        try {
            return Currency.valueOf(value);
        } catch (IllegalArgumentException ex) {
            throw new IllegalArgumentException("Unsupported currency: " + value);
        }
    }
}

File: services/payments-service/src/main/kotlin/pl/xsware/payments/domain/model/Currency.kt
package pl.xsware.payments.domain.model

enum class Currency {
    PLN,
    EUR,
    USD;

    companion object {
        fun from(value: String): Currency =
            try {
                valueOf(value)
            } catch (ex: IllegalArgumentException) {
                throw IllegalArgumentException("Unsupported currency: $value")
            }
    }
}
